amplify-tools/amplify-tools-gradle-plugin/src/main/groovy/com/amplifyframework/tools/gradle/plugin/AmplifyTools.groovy

```diff
@@ -5,228 +5,222 @@ import groovy.json.JsonSlurper
-					def systemDebugNode = new Node(configurationNode, 'GradleScriptDebugEnabled', null, true)
-					def methodNode = new Node(configurationNode, 'method', [v:"2"])
-
-					RunManagerNode.append(configurationNode)
-
-					//Save File
-					def writer = new FileWriter('./.idea/workspace.xml')
-
-					//Pretty print XML
-					groovy.xml.XmlUtil.serialize(xml, writer)
-				}
-			}
-		}
-
-		project.task('addAmplifyPushToWorkspace') {
-			if(project.file('./.idea/workspace.xml').exists()) {
-				//Open file
-				def xml = new XmlParser().parse('./.idea/workspace.xml')
-				def RunManagerNode = xml.component.find { it.'@name' == 'RunManager' } as Node
-				def configAmplifyPushCheck = RunManagerNode.children().find { it.'@name' == 'amplifyPush' } as Node
-
-				if (!configAmplifyPushCheck) {
-					// Nested nodes for amplifyPush run configuration
-					def configurationNode = new Node(null, 'configuration', [name: "amplifyPush", type:"GradleRunConfiguration", factoryName:"Gradle", nameIsGenerated:"true"])
-					def externalSystemNode = new Node(configurationNode, 'ExternalSystemSettings')
-					def executionOption = new Node(externalSystemNode, 'option', [name: "executionName"])
-					def projectPathOption = new Node(externalSystemNode, 'option', [name: "externalProjectPath", value: "\$PROJECT_DIR\$"])
-					def externalSystemIdOption = new Node(externalSystemNode, 'option', [name: "externalSystemIdString", value: "GRADLE"])
-					def scriptParametersOption = new Node(externalSystemNode, 'option', [name: "scriptParameters", value: ""])
-					def taskDescriptionsOption = new Node(externalSystemNode, 'option', [name: "taskDescriptions"])
-					def descriptionList = new Node(taskDescriptionsOption, 'list')
-					def taskNamesOption = new Node(externalSystemNode, 'option', [name: "taskNames"])
-					def nameList = new Node(taskNamesOption, 'list')
-					def amplifyPushOption = new Node(nameList, 'option', [value: "amplifyPush"])
-					def vmOption = new Node(externalSystemNode, 'option', [name: "vmOptions", value: ""])
-					def systemDebugNode = new Node(configurationNode, 'GradleScriptDebugEnabled', null, true)
-					def methodNode = new Node(configurationNode, 'method', [v:"2"])
-
-					RunManagerNode.append(configurationNode)
-
-					//Save File
-					def writer = new FileWriter('./.idea/workspace.xml')
-
-					//Pretty print XML
-					groovy.xml.XmlUtil.serialize(xml, writer)
-				}
-			}
-		}
-	}
+    void apply(Project project) {
+        // profile name can be changed in amplify-gradle-config
+        def profile = 'default'
+        def accessKeyId = null
+        def secretAccessKey = null
+        def region = null
+        def envName = 'amplify'
+        def syncEnabled = 'true'
+        def gradleConfigFileName = 'amplify-gradle-config.json'
+
+        project.task('createAmplifyApp') {
+            def npx = 'npx'
+
+            if (project.file(gradleConfigFileName).exists()) {
+                return
```

Comment:
Should we update this file, across multiple plugin runs, in case the project config has changed in between? Either way, is it warranted to log an Info for this condition?

Reply:
I don't think we want to run it each time. Since this runs `amplify-app`, it could conceivably clobber other state. I'm also slightly skeptical on logging as I think this will be noise as first time users won't encounter this state. I think!